internal/transport/http2_server.go

```diff
@@ -642,8 +642,12 @@ func (t *http2Server) HandleStreams(ctx context.Context, handle func(*Stream)) {
 		switch frame := frame.(type) {
 		case *http2.MetaHeadersFrame:
 			if err := t.operateHeaders(ctx, frame, handle); err != nil {
-				t.Close(err)
-				break
+				t.controlBuf.put(&goAway{
```

Comment:
Also, please add a comment in here to mention that an error from operateHeaders is when this invalid stream ID situation happens.

Reply:
I'd phrase it as: "Any error processing client headers, e.g. invalid stream ID, is considered a protocol violation."